sphinx/writers/html.py

```diff
@@ -75,9 +75,15 @@ def translate(self) -> None:
         self.clean_meta = ''.join(self.visitor.meta[2:])
 
 
-class HTMLTranslator(SphinxTranslator, BaseTranslator):
+if TYPE_CHECKING:
+    _MIXIN_BASE = BaseTranslator
+else:
+    _MIXIN_BASE = object
+
+
+class HTMLTranslatorCommonMixin(SphinxTranslator, _MIXIN_BASE):
     """
-    Our custom HTML translator.
+    Overrides that are common to both HTMLTranslator and HTML5Translator.
```

Comment:
It seems this is a different topic from this PR. It should be discussed in another one. I feel this makes HTML writers strongly coupled. Then it makes removing the HTML4 writer from core difficult.

Reply:
Ah, sorry. I found this as https://github.com/sphinx-doc/sphinx/pull/10250.